plasmapy/formulary/relativity.py

```diff
@@ -150,3 +155,84 @@ def relativistic_energy(m: u.kg, v: u.m / u.s) -> u.Joule:
     """
     γ = Lorentz_factor(v)
     return γ * m * c ** 2
+
+
+class RelativisticBody:
+    """
+    A physical object that is moving a speed.
+
+    Parameters
+    ----------
+    particle : |ParticleLike|, optional
+
+    quantity : |Quantity|
+        A |Quantity| with units of speed, energy,
+
+    Z : integer, optional
+
+    mass_numb : integer, optional
+    """
+
+    @validate_quantities
+    def __init__(
+        self,
+        particle=None,
+        *,
+        kinetic_energy: u.J = None,
+        Z: Integral = None,
+        mass_numb: Integral = None,
+    ):
+
+        self._data = {
+            "particle": _physical_particle_factory(particle, Z=Z, mass_numb=mass_numb)
+        }
+        self.kinetic_energy = kinetic_energy
+
+    @property
+    def particle(self) -> Union[CustomParticle, Particle, ParticleList]:
+        print(self._data)
+        return self._data["particle"]
+
+    @particle.setter
+    def particle(self, particle: ParticleLike):
+        self._data["particle"] = _physical_particle_factory(particle)
+
+    @property
+    def rest_mass(self) -> u.Quantity[u.kg]:
+        return self.particle.mass
+
+    @property
+    def mass_energy(self) -> u.Quantity[u.J]:
+        """The rest mass energy, :math:`m_0 c^2`."""
+        return self.rest_mass * c ** 2
+
+    @property
+    def total_energy(self) -> u.Quantity[u.J]:
```

Comment:
Would you also want a setter for that one?

Reply:
Eventually!  Right now I'm planning to have setters for the Lorentz factor, total energy, velocity, and the lightspeed fraction.